Rations!

FalloutFalcon opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Adds a whole bunch of highly varied rations to replace the common beans, donks, and rationpacks that litter most ships
https://hackmd.io/THabcH5VRuuHCBjSV9iJgQ

Sprites were done by stiel!
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Fixes weird lore stuff about why donk pockets would be on none syndicate ships and helps give edible food to species with limited diets + everyone hates rationpacks

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

:cl:
add: better rations
del: rationpack spawns
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

FalloutFalcon committed:
yippee

FalloutFalcon committed:
yippee

FalloutFalcon committed:
2nd ration

FalloutFalcon committed:
5/20 rations c:

FalloutFalcon committed:
better desc mostly

FalloutFalcon committed:
we dont have seafood oops

FalloutFalcon committed:
replacing rationpacks + some donks

FalloutFalcon committed:
oops

FalloutFalcon committed:
refrences

FalloutFalcon committed:
mmmm now i want a crab rangoon

FalloutFalcon committed:
its been opened

Erikafox commented on code/modules/cargo/packs/food.dm:
```diff
@@ -245,3 +245,16 @@
+/datum/supply_pack/food/ration
+	name = "Ration Crate"
+	desc = "6 standerd issue rations."
+	cost = 3000
```
3000 is a bit expensive. For context, you could buy 3 winchester rifles for 2000

FalloutFalcon replied:
I'm comparing it to pizza which is 6000. I think that's a greater issue that most of the crates are overpriced or the guns are too cheap

        name = "Donk Pocket Variety Crate"
	desc = "Featuring a line up of Donk Co.'s most popular pastry!"
	cost = 2000

	name = "Pizza Crate"
	desc = "Best prices on this side of the galaxy. All deliveries are guaranteed to be 99% anomaly-free!"
	cost = 6000 // Best prices this side of the galaxy.

Erikafox replied:
yeah food doesn't need to be that expensive.

FalloutFalcon replied:
name = "Cat Crate"
	desc = "The cat goes meow! Comes with a collar and a nice cat toy! Cheeseburger not included."//i can't believe im making this reference
	cost = 5000 //Cats are worth as much as corgis.

Erikafox replied:
you said the issue is that crates are overpriced. You don't need to contribute to the issue.

FalloutFalcon replied:
dropping it to 2k will fix some of the other prices in a later pr

Erikafox replied:
I would say 600-1200 myself.

FalloutFalcon replied:
I would say thats low balling it since the rations are ALOT of food

Erikafox commented on code/modules/food_and_drinks/drinks/drinks.dm:
```diff
@@ -758,3 +758,39 @@
 	list_reagents = list(/datum/reagent/medicine/molten_bubbles/plasma = 50)
+
+/obj/item/reagent_containers/food/drinks/ration
+	name = "Rationed Air"
```
why not like. "Generic Ration Drink". Rationed air seems to re-occur as a name later.

FalloutFalcon replied:
I forgor to change it but it is not meant to show up in game anyway
Will change

Erikafox commented on code/modules/food_and_drinks/food/condiment.dm:
```diff
@@ -309,6 +305,45 @@
+/obj/item/reagent_containers/food/condiment/pack/fruit_smoothie_powder
+	name = "tropical blend fruit and vegetable smoothie powder pack"
+	originalname = "tropical blend fruit and vegetable moothie powder"
+	list_reagents = list(/datum/reagent/consumable/sugar = 3, /datum/reagent/consumable/applejuice = 2, /datum/reagent/consumable/orangejuice = 2, /datum/reagent/consumable/eggyolk = 3, )
```
need waterflasks - waterskins tbh.

FalloutFalcon replied:
Uh contact valorium about this one

Valorium-RB replied:
The idea is that you put the powder and the water into a mess kit cup (which is standard practice) and mix them, rather than coming with a plastic pouch. Sometimes they do, but not always, especially for drinks like coffee which are intended to be consumed hot.

Erikafox commented on code/modules/reagents/chemistry/reagents/food_reagents.dm:
```diff
@@ -849,3 +849,50 @@
+	taste_mult = 2
+	taste_description = "cheese"
+
+/datum/reagent/waterpowder
```
I'm questioning the logic of dehydrated water.

MarkSuckerberg replied:
LMAO

FalloutFalcon replied:
Um contact valorium about this one

Valorium-RB replied:
Dehydrated water isn't in the design doc? Why do you have it in here

FalloutFalcon replied:
removed any mention of waterpowder and replaced it with water

FalloutFalcon committed:
addresing reviews

FalloutFalcon committed:
tiny fix

AlexanderM-Dev0 committed:
Mixable Drink Rations overhaul

Replaces "Dehydrated water" with a bag of actual water. Also Fixes recipes for mixing drinks.

AlexanderM-Dev0 committed:
Updates rations.dm

Adds more ration types with snacks, condiments and lukewarm beverage bags

Valorium-RB commented on code/modules/food_and_drinks/food/snacks_other.dm:
```diff
@@ -761,3 +761,214 @@
+
+/obj/item/reagent_containers/food/snacks/ration
+	name = "Rationed Air"
+	desc = "standerd issue ration"
```
Standard*

AlexanderM-Dev0 committed:
Updates to Rations

Updated Rations to have better names, added a new place for the remaining condiments and then added declarations for snacktypes

Valorium-RB commented on code/modules/food_and_drinks/food/snacks_other.dm:
```diff
@@ -761,3 +761,698 @@
+	list_reagents = list(/datum/reagent/consumable/nutriment = 5)
+	filling_color = "#b82121"
+	tastes = list("fish" = 1, "fried" = 1)
+	foodtype = FRIED
```
This should be SEAFOOD as well.

FalloutFalcon replied:
small issue. Seafood is not real! i can add it but i also need the taste preferences all the species should have for it

FalloutFalcon replied:
That seems like a seperate pr tbh

Valorium-RB commented on code/modules/food_and_drinks/food/snacks_other.dm:
```diff
@@ -761,3 +761,698 @@
+    tastes = list("vegetable" = 1, "crackers" = 1)
+    foodtype = GRAIN
+
+/obj/item/reagent_containers/food/snacks/ration/teriyaki_beef_sticks
```
Is there a reason the teriyaki beef sticks found in two of the rations are different items? They should probably just be the same.

FalloutFalcon replied:
The definition of insanity. Fixing it

Valorium-RB commented on code/modules/food_and_drinks/food/snacks_other.dm:
```diff
@@ -761,3 +761,698 @@
+    list_reagents = list(/datum/reagent/consumable/nutriment = 5)
+    filling_color = "#336699"
+    tastes = list("fish" = 1)
+    foodtype = MEAT
```
Also SEAFOOD.

Valorium-RB commented on code/modules/food_and_drinks/food/snacks_other.dm:
```diff
@@ -761,3 +761,698 @@
+
+/obj/item/reagent_containers/food/snacks/ration/channeler_meat_candy
+    name = "channeler meat candy"
+    desc = "An otherworldly treat made from channeler meat, offering an unusual and captivating flavor experience."
```
I'd like this to read "A traditional meat-candy from the Antechannel League on Kalixcis, offering an unusual and captivating flavor experience."

FalloutFalcon replied:
done

FalloutFalcon committed:
fixed that i think

FalloutFalcon committed:
cleanong up

FalloutFalcon committed:
cleanup

FalloutFalcon committed:
Merge branch 'master' of https://github.com/shiptest-ss13/Shiptest into rations

FalloutFalcon committed:
fixes...

Signed-off-by: FalloutFalcon <86381784+FalloutFalcon@users.noreply.github.com>

FalloutFalcon committed:
laptop commit prep for sprite

FalloutFalcon committed:
sprites and testing ration heater

FalloutFalcon committed:
mm add last 3 and change condi

FalloutFalcon committed:
flameless ration heater

Zevotech commented:
this is really cool and i hope this gets merged so i can inteq ration RP

FalloutFalcon committed:
ough nearly done

FalloutFalcon committed:
ready for review!!!!

FalloutFalcon committed:
replaces rationpacks from new ships

FalloutFalcon committed:
Merge branch 'master' into rations

MarkSuckerberg commented on code/modules/food_and_drinks/food/ration.dm:
```diff
@@ -0,0 +1,767 @@
+/obj/item/reagent_containers/food/snacks/ration
+	name = "nutriment ration"
+	desc = "standerd issue ration"
```
```suggestion
	desc = "standard issue ration"
```

MarkSuckerberg commented on code/modules/food_and_drinks/kitchen_machinery/microwave.dm:
```diff
@@ -358,6 +358,48 @@
+			visible_message("<span class='notice'>\The [target] rapidly begins cooking...</span>")
+			playsound(src, 'sound/items/cig_light.ogg', 50, 1)
+			moveToNullspace()
+
```
```suggestion
/obj/item/ration_heater/afterattack(atom/target, mob/user, flag)
	if(istype(target, /obj/item/reagent_containers/food) || istype(target, /obj/item/grown))
		to_chat(user, "<span class='notice'>You start sliding \the [src] under the [target]...</span>")
		if(do_after(user, 10))
			tocook = target
			RegisterSignal(tocook, COMSIG_PARENT_QDELETING, PROC_REF(clear_cooking))
			target.add_overlay(ration_overlay)
			addtimer(CALLBACK(src, .proc/cook), 100)
			visible_message("<span class='notice'>\The [target] rapidly begins cooking...</span>")
			playsound(src, 'sound/items/cig_light.ogg', 50, 1)
			moveToNullspace()

/obj/item/ration_heater/proc/clear_cooking(datum/source)
	SIGNAL_HANDLER
	UnregisterSignal(tocook, COMSIG_PARENT_QDELETING)
	tocook.cut_overlay(ration_overlay)
	tocook = null
```
I want to make sure this doesn't cause harddels

MarkSuckerberg commented on code/modules/food_and_drinks/kitchen_machinery/microwave.dm:
```diff
@@ -358,6 +358,48 @@
+	playsound(tocook, 'sound/items/cig_snuff.ogg', 50, 1)
+	if(istype(tocook, /obj/item/reagent_containers/food) || istype(tocook, /obj/item/grown))
+		tocook.cut_overlay(ration_overlay)
+		tocook.microwave_act()
```
```suggestion
		clear_cooking()
		tocook.microwave_act()
```

MarkSuckerberg commented on code/modules/reagents/chemistry/reagents/food_reagents.dm:
```diff
@@ -849,3 +849,22 @@
+/datum/reagent/waterpowder
+	name = "Instant Water Powder"
+	description = "Activates into steaming hot water"
+	reagent_state = SOLID
```
```suggestion
```

MarkSuckerberg reviewed (changes requested):
Just a few final things

FalloutFalcon committed:
Update code/modules/reagents/chemistry/reagents/food_reagents.dm

Forgot to remove that

Co-authored-by: Mark Suckerberg <mark@suckerberg.gay>
Signed-off-by: FalloutFalcon <86381784+FalloutFalcon@users.noreply.github.com>

FalloutFalcon committed:
Update code/modules/food_and_drinks/kitchen_machinery/microwave.dm

I dont fully understand the diffrent between deletes so i will trust you on this on

Co-authored-by: Mark Suckerberg <mark@suckerberg.gay>
Signed-off-by: FalloutFalcon <86381784+FalloutFalcon@users.noreply.github.com>

FalloutFalcon committed:
Update code/modules/food_and_drinks/food/ration.dm

spelling!

Co-authored-by: Mark Suckerberg <mark@suckerberg.gay>
Signed-off-by: FalloutFalcon <86381784+FalloutFalcon@users.noreply.github.com>

FalloutFalcon committed:
Update code/modules/food_and_drinks/kitchen_machinery/microwave.dm

Co-authored-by: Mark Suckerberg <mark@suckerberg.gay>
Signed-off-by: FalloutFalcon <86381784+FalloutFalcon@users.noreply.github.com>

FalloutFalcon committed:
Merge branch 'master' into rations
